src/omt/omt_optimizer.cpp

```diff
@@ -39,32 +42,143 @@ std::unique_ptr<OMTOptimizer> OMTOptimizer::getOptimizerForNode(TNode targetNode
   else if (objectiveType.isBitVector())
   {
     // bitvector type: use OMTOptimizerBitVector
-    return std::unique_ptr<OMTOptimizer>(new OMTOptimizerBitVector(isSigned));
+    return std::unique_ptr<OMTOptimizer>(
+        new OMTOptimizerBitVector(objective.bvIsSigned()));
   }
   else
   {
     return nullptr;
   }
 }
 
-std::unique_ptr<SmtEngine> OMTOptimizer::createOptCheckerWithTimeout(
-    SmtEngine* parentSMTSolver, bool needsTimeout, unsigned long timeout)
+Node OMTOptimizer::mkStrongIncrementalExpression(
+    NodeManager* nm, TNode lhs, TNode rhs, OptimizationObjective& objective)
+{
+  constexpr const char lhsTypeError[] =
+      "lhs type does not match or is not implicitly convertable to the target "
+      "type";
+  constexpr const char rhsTypeError[] =
+      "rhs type does not match or is not implicitly convertable to the target "
+      "type";
+  TypeNode targetType = objective.getTarget().getType();
+  switch (objective.getType())
+  {
+    case OptimizationObjective::MINIMIZE:
+    {
+      if (targetType.isInteger())
+      {
+        Assert(lhs.getType().isInteger() || lhs.getType().isReal())
+            << lhsTypeError;
+        Assert(rhs.getType().isInteger() || rhs.getType().isReal())
+            << rhsTypeError;
+        return nm->mkNode(Kind::LT, lhs, rhs);
+      }
+      else if (targetType.isBitVector())
+      {
+        Assert(lhs.getType() == targetType) << lhsTypeError;
+        Assert(rhs.getType() == targetType) << rhsTypeError;
+        return (objective.bvIsSigned())
+                   ? (nm->mkNode(Kind::BITVECTOR_SLT, lhs, rhs))
+                   : (nm->mkNode(Kind::BITVECTOR_ULT, lhs, rhs));
+      }
+      else
+      {
+        CVC5_FATAL() << "Target type does not support optimization";
```

Comment:
All `CVC5_FATAL` should be replaced with `Unreachable` because it is used in a code that is supposed to be unreachable (this is also guarded by assertions above). 

So please use Unreachable instead for all occurrences. The error message is very useful though, so please still use it.

Reply:
Unreachable doesn't tell you the reasons. 
and Unreachable() is CVC5_FATAL() << "Unreachable code reached"...

Reply:
Right, that's why I ask that you keep the error message so that we see both the reason and that it is unreachable.

Reply:
I wouldn't say it's unreachable, because it could be reached logically if you pass other types to it. 
just a CVC5_FATAL() with an error message is more desired. 
Unreachable are used to tag really unreachable code AFAIK.

Reply:
How about `Unimplemented()` then?

Reply:
I don't plan to support FLOATINGPOINT TBH...